homeassistant/components/risco/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The Risco integration."""
+import asyncio
+
+from pyrisco import RiscoAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_PIN, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS = ["alarm_control_panel"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Risco component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Risco from a config entry."""
+    data = entry.data
+    risco = RiscoAPI(data[CONF_USERNAME], data[CONF_PASSWORD], data[CONF_PIN])
+    await risco.login()
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = risco
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        risco = hass.data[DOMAIN].pop(entry.entry_id)
+        await risco.close()
```

Comment:
It doesn't look like this is called when shutting HA down. I'm seeing:
```
Error doing job: Unclosed client session
Error doing job: Unclosed connector
```

Not sure how to handle.

Reply:
Sounds like a client session isn't closed properly in the package upstream?
Alternatively, adjust the client library to accept a session, and pass in Home Assistant' session into the instance creation of risico.

Reply:
Thanks.
Calling `close()` in console sessions works correctly, so I'm not sure the problem is upstream, but I can look into having it accept a session.
Where would I get Home Assistant's session?

Reply:
BTW I also added prints to `async_unload_entry` and it doesn't look like it's being called during shutdown at all.

Reply:
Resolved in https://github.com/home-assistant/core/pull/36930/commits/f5ed9aea71164618202346ff13a3c9bf3b5c7a6c